org.eclipse.m2e.core/src/org/eclipse/m2e/core/internal/project/ProjectConfigurationManager.java

```diff
@@ -500,9 +508,62 @@ private void updateProjectConfiguration(ProjectConfigurationRequest request, IPr
       return null;
     }, monitor);
 
+    // must happen after calling the project configurators (as it checks the set Java project JRE)
+    configurationRequiredJavaVersion(mavenProjectFacade, monitor);
     log.debug("Updated project configuration for {} in {} ms.", mavenProjectFacade, System.currentTimeMillis() - start); //$NON-NLS-1$
   }
 
+  /**
+   * Configure Java version required for Maven execution based on maven-enforcer-plugin configuration
+   */
+  private void configurationRequiredJavaVersion(IMavenProjectFacade facade, IProgressMonitor monitor) {
```

Comment:
I still would prefer to move that to the `MavenLaunchDelegate`.
This would save the logic to persist that value and unless there is a strong need to make that available in the API (see my other comment), I don't think this is necessary here. Reading the value is not very costly.